### ISSUE TITLE
[FR] Support for G10 L2(0) P1-6 X0 Y0 Z0 work coordinates (G54-G59)

### ISSUE THREAD

**Issue by telleropnul** [MarlinFirmware/Marlin#14734]
[marlin-bugfix-2.0]

Thank you for all your hard work developing and maintaining the Marlin codebase!

Marlin has support for CNC coordinate systems, which is most appreciated.

[CNC_Coordinate_Systems] can be enabled in [configuration_adv.h] to enable G53-G59 coordinate systems:

```
/**
* CNC Coordinate Systems
*
* Enables G53 and G54-G59.3 commands to select coordinate systems
* and G92.1 to reset the workspace to native machine space.
*
#define CNC_COORDINATE_SYSTEMS
```

Marlin is increasingly used for CNC machines i.e. MPCNC.

Popular Gcode parsers like CNC.js and LinuxCNC use "G10 L20 P1 X0" and "G10 L20 P1 Y0" and "G10 L20 P1 Z0" -or simply "G10 L20 P1 X0 Y0 Z0"- to set a new origin for G54 work (offset) coordinates relative to the current position.

Marlin already supports G53 machine coordinates and G54-G59 work (offset) coordinates but unfortunately does not support G10 L20 P1 X0 Y0 Z0 as of yet.

A more detailed explanation:
https://github.com/synthetos/g2/wiki/Gcode-Coordinate-Offsets

For example, G10 L20 P**1** X0 Y0 simply requires Marlin to execute the following:

1. **G54** ; P is coordinate system 1-6, corresponding to G54 - G59, respectively
2. G92 X0 Y0
3. Return to original coordinate system.

For example, G10 L20 P**2** X0 Y0 simply requires Marlin to execute the following:
1. **G55**
2. G92 X0 Y0
3. Return to original coordinate system.

Etc.

Note: Caution - Marlin also uses G10 to control 3D printer retraction.  Adding a 'G10 L20' option to the existing 'G10' command should not affect retraction behavior.

**Comment by fantasmag0ria** [comment]
Hi, @telleropnul is it kinda what you had in mind?

https://github.com/cncjs/cncjs/issues/549

**Comment by telleropnul** [comment]
Thank you very much for looking into this.

This ticket is now closed and progress is tracked in cncjs/cncjs#549
